### DISCUSSION TITLE
Parsing a subset of the ast tree and making sure all imported and/or referenced subtrees are included

### DISCUSSION THREAD

**Discussion by ymolists** [cue-lang/cue#2781] [Q&A]
Hi folks

The tool i ma writing is trying to lift the types in cue and generate ts, python code. Unfortunately from what i can see i am forced to use ast walking. For example listlit only allows iterating on concrete values not types of incomplete types.

As such, I am trying to parse the ast tree formed by the snippet of code below.  The problem is that i dont want to handle types references and/or imports while doing ast traversals. Is it possible to form a full tree from say node `s00` and then do the traversal ? I am looking at `ast.Resolve` in the hopes that it does what i want it to do.

```
-- cue.mod/module.cue --
module: "example.com"

-- main.cue --
package test

import "example.com/dep"

dep

s00:{
  dep
}

-- dep/file.cue --
package dep

empty: int
```

**Comment by myitcv** [comment]
> Unfortunately from what i can see i am forced to use ast walking.

If you are looking to generate anything from CUE, you almost certainly want to walk the `cue.Value` and not the AST. 

> For example listlit only allows iterating on concrete values not types of incomplete types.

This is all possible via the `cue.Value` API:

```
go mod tidy
go run .
cmp stdout stdout.golden

-- go.mod --
module mod.com

go 1.21.4

require cuelang.org/go v0.7.0

-- main.go --
package main

import (
	"fmt"

	"cuelang.org/go/cue"
	"cuelang.org/go/cue/cuecontext"
)

const src = `
x: [string, ...int]
`

func main() {
	ctx := cuecontext.New()
	v := ctx.CompileString(src)
	x0 := v.LookupPath(cue.ParsePath("x[0]"))
	xother := v.LookupPath(cue.MakePath(cue.Str("x"), cue.AnyIndex))

	fmt.Printf("x[0]: %v\n", x0)
	fmt.Printf("x[1:]: %v\n", xother)
}
-- stdout.golden --
x[0]: string
x[1:]: int
```
